TicTacToe/TicTacToe.cs

```diff
@@ -29,51 +32,141 @@ public static void GetInput()
         {
             Console.WriteLine("Player " + playerTurn);
             Console.WriteLine("Enter Row:");
-            int row = int.Parse(Console.ReadLine());
+            
+            string input = Console.ReadLine();
+            int row = 0;
+            // limits input to 0, 1, 2
+            if(input == "0" || input == "1" || input == "2")
+            {
+                row = int.Parse(input);
+            }
+            else
+            {
+                Console.WriteLine("Invalid Input, Try Again");
+                GetInput();
+            }
+                
+
             Console.WriteLine("Enter Column:");
-            int column = int.Parse(Console.ReadLine());
+            string inputtwo = Console.ReadLine();
+            int column = 0;
+            if(inputtwo == "0" || inputtwo == "1" || inputtwo == "2")
+            {
+              column = int.Parse(inputtwo);
+            }
+            else
+            {
+                Console.WriteLine("Invalid Input, Try Again");
+                GetInput();
+            }
+            
+            PlaceMark(row,column);
+             if (playerTurn == "O")
+            {
+                playerTurn = "X";
+            }
+            else 
+            {
+                playerTurn ="O";
+            }
+            
+            DrawBoard();
         }
-
+        /*update board with correct value and position */
         public static void PlaceMark(int row, int column)
         {
+            board[row][column] = playerTurn;
         // your code goes here
+    
+
         }
 
         public static bool CheckForWin()
         {
             // your code goes here
+            bool won = false;
+            if(HorizontalWin() || VerticalWin() || DiagonalWin())
+            {
+                won = true;
+                Console.Write("You Won!");
+            }
 
-            return false;
+            return won;
         }
 
         public static bool CheckForTie()
         {
-            // your code goes here
-
-            return false;
+            // checking for Tie
```

Comment:
This is not really checking for a tie, its just checking that every position is played.